cmd/contact-exporter/main.go

```diff
@@ -69,26 +71,46 @@ func writeContacts(contactsList []contact, outfile string) error {
 const usageIntro = `
 Introduction:
 
-The contact exporter exists to retrieve the email addresses of all registered
-users with currently unexpired certificates. This list of email addresses can
+The contact exporter exists to retrieve the IDs of all registered
+users with currently unexpired certificates. This list of registration IDs can
 then be given as input to the notification mailer to send bulk notifications.
 
-The email addresses are deduplicated and sorted prior to being writen to the
-outfile. E.g. if two registrations exist with the contact email
-"example@example.com", this address will only appear once. Registration contacts
-that are *not* email addresses are discarded (e.g. tel:999-999-9999)
+The -grace parameter can be used to allow registrations with certificates that
+have already expired to be included in the export. The argument is a Go duration
+obeying the usual suffix rules (e.g. 24h).
+
+Registration IDs are favoured over email addresses as the intermediate format in
+order to ensure the most up to date contact information is used at the time of
+notification. The notification mailer will resolve the ID to email(s) when the
+mailing is underway, ensuring we use the correct address if a user has updated
+their contact information between the time of export and the time of
+notification.
+
+The contact exporter's registration ID output will be JSON of the form:
+  [
+   { "id": 1 },
+   ...
+   { "id": n }
+  ]
 
 Examples:
-  Export all email addresses to "emails.txt":
+  Export all registration IDs with unexpired certificates to "regs.json":
+
+  contact-exporter -config test/config/contact-exporter.json -outfile regs.json
+
+  Export all registration IDs with certificates that are unexpired or expired
+  within the last two days to "regs.json":
 
-  contact-exporter -config test/config/contact-exporter.json -outfile emails.txt
+  contact-exporter -config test/config/contact-exporter.json -grace 48h -outfile
+    "regs.json"
 
 Required arguments:
 - config
 - outfile`
 
 func main() {
 	outFile := flag.String("outfile", "", "File to write contacts to (defaults to stdout).")
+	grace := flag.Duration("grace", 2*24*time.Hour, "Include contacts with certificates that expired in < grace ago")
```

Comment:
@rolandshoemaker Hmmmm. Suggestions on a fix? I could make it a `flag.Int` but that will make the case where you want to specify a non-zero grace period awkward.